File: backend/bluffword-backend/src/main/java/org/bluffwordbackend/models/Round.java
package org.bluffwordbackend.models;

import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString;

import java.util.HashSet;
import java.util.Set;

@Entity
@Data
public class Round {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private int roundNumber;
    private String category;

    @ManyToOne(optional = false)
    @JoinColumn(name = "game_room_id")
    @ToString.Exclude
    @EqualsAndHashCode.Exclude
    private GameRoom gameRoom;

    @OneToMany(mappedBy = "round")
    @ToString.Exclude
    @EqualsAndHashCode.Exclude
    private Set<PlayerRound> playerRounds = new HashSet<>();

    @OneToMany(mappedBy = "round")
    @ToString.Exclude
    @EqualsAndHashCode.Exclude
    private Set<Vote> votes = new HashSet<>();

    @ManyToOne(optional = false)
    @JoinColumn(name = "word_pair_id")
    private WordPair wordPair;

}

File: backend_kotlin/src/main/kotlin/org/project/backend_kotlin/redisModels/Round.kt
package org.project.backend_kotlin.redisModels

data class Round(
    val roundNumber: Int,
    val category: String,
    //key: playerId, value: answer
    val listOfAnswers: MutableMap<String, String>,
    val wordPair: WordPair,
    //key: voterId, value: targetId
    val votes: MutableMap<String, String>
)
